### ISSUE TITLE
timeout not supported in StartRegistrationOptions?

### ISSUE THREAD

**Issue by kasecato** [Yubico/java-webauthn-server#23]
Hello, first of all, thank you for your great library.

I'd like to know about `StartRegistrationOptions` and `StartAssertionOptions` builder does not have a timeout interface.

Thanks

- [RelyingParty.java#L284](https://github.com/Yubico/java-webauthn-server/blob/master/webauthn-server-core/src/main/java/com/yubico/webauthn/RelyingParty.java#L284)
- [StartRegistrationOptions.java](https://github.com/Yubico/java-webauthn-server/blob/master/webauthn-server-core/src/main/java/com/yubico/webauthn/StartRegistrationOptions.java)
- [StartAssertionOptions.java](https://github.com/Yubico/java-webauthn-server/blob/master/webauthn-server-core/src/main/java/com/yubico/webauthn/StartAssertionOptions.java)
- [5.4. Options for Credential Creation (dictionary PublicKeyCredentialCreationOptions)](https://w3c.github.io/webauthn/#dictionary-makecredentialoptions)
- [5.5. Options for Assertion Generation (dictionary PublicKeyCredentialRequestOptions)](https://w3c.github.io/webauthn/#dictionary-assertion-options)

**Comment by emlun** [comment]
Hi! Thanks for the report and kind words. This is an oversight on my part, will fix.

**Comment by emlun** [comment]
In the meantime, you can work around this by setting the timeout yourself on the returned `PublicKeyCredentialCreationOptions` and `PublicKeyCredentialRequestOptions`:

```
PublicKeyCredentialCreationOptions pkcco = rp.startRegistration(registrationOptions)
    .toBuilder()
    .timeout(1337)
    .build();

AssertionRequest tmp = rp.startAssertion(assertionOptions);
AssertionRequest assertReq = tmp
    .toBuilder()
    .publicKeyCredentialRequestOptions(
        tmp.getPublicKeyCredentialRequestOptions()
            .toBuilder()
            .timeout(1337)
            .build()
    )
    .build();
```

Not very pretty, but functionally equivalent.

**Comment by kasecato** [comment]
That’s wonderful!
Thanks 🎉

**Comment by emlun** [comment]
Fixed in version 1.3.0-RC1 which should show up on Maven Central soon. If nothing else comes up in about 2 weeks, we'll promote that to the full 1.3.0 release. Thanks again!
